errorHelpers.go

```diff
@@ -22,31 +22,53 @@ type jsonError []struct {
 }
 
 type xmlError struct {
-	Message string `xml:"Body>Fault>faultstring"`
+	Message   string `xml:"Body>Fault>faultstring"`
 	ErrorCode string `xml:"Body>Fault>faultcode"`
 }
 
+type SalesforceError struct {
+	Message      string
+	HttpCode     int
+	ErrorCode    string
+	ErrorMessage string
+}
+
+func (err SalesforceError) Error() string {
+	return err.Message
```

Comment:
errMessage is always present, while ErrorMessage is present only when the error was successfully parsed.

This method can be made smarter, to return succinct message when available, like

```
if err.ErrorMessage != "" {
  return err.ErrorMessage
} else {
  return err.Message
}
```

but that would change previous behavior. Please let me know what you think